### PR TITLE
gh-118392: Add note about random.random for multi thread app

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-118392

----
📚 Documentation preview 📚: https://cpython-previews--118396.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/random.rst [hunk 1]
```diff
@@ -294,6 +294,10 @@ be found in any statistics text.
 
    Return the next random floating point number in the range ``0.0 <= X < 1.0``
 
+   .. warning::
+      Calling this function in the concurrent use can cause performance degradation.
+      Instead, Consider using :class:`Random` per thread.
+
```
Comment (colesbury) [inline Doc/library/random.rst]: Let's move this up to the module level, perhaps between the security warning and the "see also". The advice is not limited this function.

I think maybe a `.. note::` is more appropriate. A red "warning" seems a bit severe to me.

For the text, something like:

```
The global random number generator and instances of :class:`Random` are thread-safe. However, 
in the free-threaded build, concurrent calls to the global generator or to the same instance of :class:`Random`
may encounter contention and poor performance. Consider using separate instances of :class:`Random`
per thread instead.
```

File: Doc/library/random.rst [hunk 2]
```diff
@@ -294,6 +294,10 @@ be found in any statistics text.
 
    Return the next random floating point number in the range ``0.0 <= X < 1.0``
 
+   .. warning::
+      Calling this function in the concurrent use can cause performance degradation.
```
Comment (terryjreedy) [inline Doc/library/random.rst]: `in the concurrent use` is not grammatical.  Either `in the concurrent use case` or `in concurrent uses` or `in multiple threads`

**Review by terryjreedy — CHANGES_REQUESTED** [review]

File: Doc/library/random.rst (hunk 1 above)
Comment (terryjreedy) [reply Doc/library/random.rst]: I agree.  This replaces my comment.

File: Doc/library/random.rst (hunk 2 above)
Comment (terryjreedy) [reply Doc/library/random.rst]: Sam's suggestion is even better.

**Comment by corona10** [comment]
I have made the requested changes; please review again

**Review by rhettinger — APPROVED** [review]

**Pull request merged**